frontend/pages/create-edit-user.js

```diff
@@ -0,0 +1,367 @@
+                  },
+                  {
+                    traitType: "string",
+                    key: "instagram",
+                    value: body?.instagram || null,
+                  },
+                  {
+                    traitType: "string",
+                    key: "discord",
+                    value: body?.discord || null,
+                  },
+                  {
+                    traitType: "string",
+                    key: "github",
+                    value: body?.github || null,
+                  },
+                ],
+                createdOn: new Date(),
+                app_id: "IPFonts",
+              };
+              const lensResponse = await fetch("./api/user-profile-data", {
+                method: "POST",
+                body: JSON.stringify(lensBody),
+              });
+              if (lensResponse.status !== 200) {
+                alert(
+                  "Oops! Something went wrong. Please refresh and try again."
+                );
+                resetFunction();
+              } else {
+                const lensResJSON = await lensResponse.json(),
+                  lensResJSONurl = `https://ipfs.io/ipfs/${lensResJSON.cid}/data.json`,
+                  getProfile = await lensClient.query({
+                    query: getProfileByHandle,
+                    variables: { handle: lensHandle },
+                  }),
+                  setLensProfile = await lensClient.mutate({
+                    mutation: createSetProfileWithMetadata,
+                    variables: {
+                      request: {
+                        profileId: getProfile.data.profile.id,
+                        metadata: lensResJSONurl,
+                      },
+                    },
+                  });
+                console.log(setLensProfile);
+              }
+            }
+            alert("User updated!");
+            resetFunction();
+          }
+        }
+      } else {
+        alert("Couldn't connect contract!");
+      }
+    } catch (error) {
+      alert(
+        `Oops! Something went wrong. Please refresh and try again. ${error}`
+      );
+      resetFunction();
+    }
+  }
+
+  // Formik structure
+  const formik = useFormik({
+    initialValues: {
+      ...formValues,
+    },
+    enableReinitialize: true,
+    validationSchema: Yup.object({ ...createEditUserValidationSchema }),
+    onSubmit: (values, { resetForm }) => {
+      const setReset = () => {
+        setWaitForm(false);
+        resetForm();
+      };
+      handleCreateEditUser(values, setReset);
+    },
+  });
+
+  return (
+    <Form onSubmit={formik.handleSubmit}>
+      {!props.connected && (
+        <h5>Connect your wallet to create or edit your user</h5>
+      )}
+      {props.connected && (
+        <>
+          <h6>{lensHandle ? `Welcome ${lensHandle}` : "Welcome"}</h6>
+          {createEditUserInputs.map((field) => {
+            if (!lensHandle && field.id === "handle") {
```

Comment:
With this conditional are we saying to only render an input and error div combination of the the `createEditUserInputs` is the lens handle input? Should the conditional be `field.id !== "handle"` meaning all inputs besides the lens handle?

Reply:
This part checks if the user has a lens profile, if is true loads all the data from that profile to the form, but this part needs to be taken from a global provider and not from this local approach, that's te main reason I want to leave this as it is temporarily, so I can still test with this frontend but start focusing on refactor and at some point come back here and chop the file, consume the global provider and make it more accessible to the user. Does that make sense?

Reply:
Ah got ya 👍 I tested out the pages and looks like if you have a lens profile the welcome message above is the only thing that is rendered...
<img width="1242" alt="image" src="https://user-images.githubusercontent.com/761087/216597123-f5e9a73c-e5c4-4bd2-9f84-6eeef0038ed7.png">
So I think this conditional should be the following to render the form inputs...
```suggestion
            if ((lensHandle && field.id !== "handle") || !lensHandle) {
```
In other words render every field except the handle field if the user has a lens profile or render all fields if a user does not have a lens handle. Also I think once we are out of the testing phase and ready to launch the mvp we can remove the code in this file around handling creating a lens user. What do you think about adding todo comments in this file that link to a new task in our trello board to make sure we cleanup that code?